app/assets/javascripts/google-tag-manager.js

```diff
@@ -0,0 +1,15 @@
+(function() {
+  const gtmDataElement = document.getElementById("data-google-tag-manager");
+  const dataLayerJson = JSON.parse(gtmDataElement.dataset.dataLayer);
+
+  /* eslint-disable */
+  // prettier-ignore
+  (function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
+  new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
+  j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
+  'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
+  })(window,document,'script','dataLayer',gtmDataElement.dataset.containerId);
+  /* eslint-enable */
+
+  window.dataLayer = [dataLayerJson];
```

Comment:
Do we have to set the `dataLayer` attribute before loading the Google Tag Manager script as per https://developers.google.com/tag-manager/devguide#adding-data-layer-variables-to-a-page? 

![screenshot from 2018-09-12 10-08-41](https://user-images.githubusercontent.com/4335742/45397734-fdc76e80-b673-11e8-8778-34e4a846f3b7.png)

Reply:
Ah good catch, thanks!